src/main/java/org/spongepowered/api/event/cause/Cause.java

```diff
@@ -48,52 +52,211 @@
  * some blocks, but tracing this event would be too complicated and thus
  * may not be attempted.</p>
  */
-public class Cause {
+@SuppressWarnings("unchecked")
+public abstract class Cause {
```

Comment:
Then what's the benefit in having an additional subclass and adding additional complexity instead of simply wrapping List and adding an `isPresent` method? It seems to me that functionality can be completely covered by a simpler overall Cause implementation.